cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.cluster.metadata;
+
+import org.apache.iotdb.db.conf.IoTDBConstant;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.exception.metadata.PathNotExistException;
+import org.apache.iotdb.db.metadata.MManager;
+import org.apache.iotdb.db.metadata.MeasurementMeta;
+import org.apache.iotdb.db.metadata.mnode.MNode;
+import org.apache.iotdb.db.qp.physical.crud.InsertPlan;
+import org.apache.iotdb.db.service.IoTDB;
+import org.apache.iotdb.tsfile.common.cache.LRUCache;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.TimeValuePair;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class CMManager extends MManager {
+
+  private static final Logger logger = LoggerFactory.getLogger(CMManager.class);
+
+  // currently, if a key is not existed in the mRemoteMetaCache, an IOException will be thrown
+  private ReentrantReadWriteLock cacheLock = new ReentrantReadWriteLock();;
+  private LRUCache<String, MeasurementMeta> mRemoteMetaCache;
+  private MetaPuller metaPuller;
+
+  protected CMManager() {
+    super();
+    metaPuller = MetaPuller.getInstance();
+    int remoteCacheSize = config.getmRemoteSchemaCacheSize();
+    mRemoteMetaCache = new LRUCache<String, MeasurementMeta>(remoteCacheSize) {
+      @Override
+      protected MeasurementMeta loadObjectByKey(String key) throws IOException {
+        throw new IOException(key + " not found!");
```

Comment:
Throwing an exception would fill the stack trace, so how about changing it to return null?

Reply:
ok